modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -831,6 +849,108 @@ public void testSendNotificationWithSystemObjectDefinition()
 			notificationQueueEntries.get(0));
 
 		objectActionLocalService.deleteObjectAction(objectAction);
+
+		// On Commerce Order payment status update
+
+		ObjectDefinition commerceOrderObjectDefinition =
+			_objectDefinitionLocalService.fetchObjectDefinitionByClassName(
+				TestPropsValues.getCompanyId(), CommerceOrder.class.getName());
+
+		Set<String> objectActionTriggerKeys = new HashSet<>();
+
+		ListUtil.isNotEmptyForEach(
+			_objectActionTriggerRegistry.getObjectActionTriggers(
+				commerceOrderObjectDefinition.getClassName()),
+			objectActionTrigger -> objectActionTriggerKeys.add(
+				objectActionTrigger.getKey()));
+
+		Assert.assertTrue(
+			objectActionTriggerKeys.contains(
+				DestinationNames.COMMERCE_ORDER_STATUS));
+
+		Assert.assertTrue(
+			objectActionTriggerKeys.contains(
+				DestinationNames.COMMERCE_PAYMENT_STATUS));
+
+		NotificationTemplate commerceOrderNotificationTemplate =
+			notificationTemplateLocalService.addNotificationTemplate(
+				NotificationTemplateUtil.createNotificationContext(
+					TestPropsValues.getUser(),
+					commerceOrderObjectDefinition.getObjectDefinitionId(),
+					RandomTestUtil.randomString(),
+					RandomTestUtil.randomString(),
+					NotificationTemplateConstants.EDITOR_TYPE_RICH_TEXT,
+					Arrays.asList(
+						createNotificationRecipientSetting(
+							"from", "[%CURRENT_USER_EMAIL_ADDRESS%]"),
+						createNotificationRecipientSetting(
+							"fromName",
+							Collections.singletonMap(
+								LocaleUtil.US, "[%CURRENT_USER_FIRST_NAME%]")),
+						createNotificationRecipientSetting(
+							"to", "[%CURRENT_USER_EMAIL_ADDRESS%]")),
+					RandomTestUtil.randomString(),
+					NotificationConstants.TYPE_EMAIL, Collections.emptyList()));
+
+		ObjectAction commerceOrderObjectAction =
+			_objectActionLocalService.addObjectAction(
+				RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+				commerceOrderObjectDefinition.getObjectDefinitionId(), true,
+				StringPool.BLANK, RandomTestUtil.randomString(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				RandomTestUtil.randomString(),
+				ObjectActionExecutorConstants.KEY_NOTIFICATION,
+				DestinationNames.COMMERCE_PAYMENT_STATUS,
+				UnicodePropertiesBuilder.put(
+					"notificationTemplateId",
+					commerceOrderNotificationTemplate.
+						getNotificationTemplateId()
+				).build(),
+				false);
+
+		Group group = GroupTestUtil.addGroup();
+
+		CommerceCurrency commerceCurrency =
+			CommerceCurrencyTestUtil.addCommerceCurrency(
+				TestPropsValues.getCompanyId());
+
+		CommerceChannel commerceChannel = CommerceTestUtil.addCommerceChannel(
+			group.getGroupId(), commerceCurrency.getCode());
+
+		PrincipalThreadLocal.setName(TestPropsValues.getUserId());
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(TestPropsValues.getUser()));
```

Comment:
@icgsbr instead of doing this, please try to add the `PermissionCheckerMethodTestRule` in the `AggregateTestRule`.

```
@ClassRule
@Rule
public static final AggregateTestRule aggregateTestRule =
	new AggregateTestRule(
		new LiferayIntegrationTestRule(),
		PermissionCheckerMethodTestRule.INSTANCE,
		SynchronousMailTestRule.INSTANCE);
```

Reply:
Hi @carolmariaabb,

By doing the above I noticed that the test fails with:

```
com.liferay.notification.internal.type.test.EmailNotificationTypeTest > testSendNotificationWithSystemObjectDefinition FAILED
    org.junit.ComparisonFailure: expected:<[f8csof2f]@liferay.com,cc@life...> but was:<[test]@liferay.com,cc@life...>
        at org.junit.Assert.assertEquals(Assert.java:117)
        at org.junit.Assert.assertEquals(Assert.java:146)
        at com.liferay.notification.internal.type.test.EmailNotificationTypeTest._assertNotificationQueueEntry(EmailNotificationTypeTest.java:1007)
        at com.liferay.notification.internal.type.test.EmailNotificationTypeTest.testSendNotificationWithSystemObjectDefinition(EmailNotificationTypeTest.java:816)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$2.evaluate(AbstractTestRule.java:90)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$1.evaluate(TestExecutorRunnable.java:228)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at com.liferay.portal.kernel.test.rule.AbstractTestRule$1.evaluate(AbstractTestRule.java:50)
        at org.junit.rules.RunRules.evaluate(RunRules.java:20)
        at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable._execute(TestExecutorRunnable.java:307)
        at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable.run(TestExecutorRunnable.java:86)
        at java.lang.Thread.run(Thread.java:750)

1 test completed, 1 failed
There were failing tests. See the report at: file:///home/igorcosta/dev/projects/liferay-portal/modules/apps/notification/notification-test/build/reports/tests/testIntegration/index.html
```

Yet, with the previous code, the test passed. I noticed that on `ObjectActionLocalServiceTest`, despite having the `PermissionCheckerMethodTestRule.INSTANCE` in the `AggregateTestRule` they also have examples declaring the permission as I did before. See [here](https://github.com/liferay/liferay-portal/blob/ecfbee9e2d43835ff481e289ebc7e3d8e2962ae7/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java#L1318C4-L1320C49).

Reply:
@icgsbr so please change to the following:

```
String originalName = PrincipalThreadLocal.getName();
PermissionChecker originalPermissionChecker =
	PermissionThreadLocal.getPermissionChecker();

try {
	PrincipalThreadLocal.setName(TestPropsValues.getUserId());
	PermissionThreadLocal.setPermissionChecker(
		PermissionCheckerFactoryUtil.create(TestPropsValues.getUser()));
   
    ....
}
finally {
	PrincipalThreadLocal.setName(originalName);
	PermissionThreadLocal.setPermissionChecker(
		originalPermissionChecker);
}
```